src/load_fund_data.py

```diff
@@ -0,0 +1,233 @@
+        Bool: pl.Boolean,
+        DateTime: pl.Datetime,
+    }
+    return mapping.get(pa_dtype.__class__, pl.String)
+
+
+def append_metadata(file_path: str, fund_name: str, date: str) -> pl.DataFrame:
+    """
+    Reads the csv and appends fund_name and date as metadata.
+    """
+    df = pl.read_csv(file_path)
+    df = df.with_columns([pl.lit(fund_name).alias("FUND"), pl.lit(date).alias("DATE")])
+    df = df.rename(
+        {col: Utils.convert_to_screaming_snake_case(col) for col in df.columns}
+    )
+    expected_columns = ExternalFund.to_schema().columns.items()
+    for column_name, column_schema in expected_columns:
+        if column_name not in df.columns:
+            pl_dtype = pandera_to_polars_dtype(column_schema.dtype)
+            df = df.with_columns(pl.lit(None, dtype=pl_dtype).alias(column_name))
+    df = df.select([col for col, _ in expected_columns]).filter(
+        ~pl.all_horizontal(pl.all().is_null())
+    )
+    return df
+
+
+def load_data_to_db(
+    database_path: str, fund_holdings_table_name: str, df: pl.DataFrame
+) -> None:
+    """
+    Loads the DataFrame into the specified SQLite database table.
+
+    Args:
+        database_path (str): The path to the SQLite database file.
+        fund_holdings_table_name (str): The name of the table to load data into.
+        df (pl.DataFrame): The DataFrame to load.
+
+    Returns:
+        None
+
+    Raises:
+        Exception: If there is an error while loading data to the database.
+    """
+    try:
+        logger = logging.getLogger()
+        connection_string = f"sqlite:///{os.path.abspath(database_path)}"
+        df.write_database(
+            connection=connection_string,
+            table_name=fund_holdings_table_name,
+            if_table_exists="replace",
+        )
+    except Exception as e:
+        logger.error(f"Error loading data to database: {e}")
+        raise
+
+
+def process_fund_data(
+    database_path: str, input_data_path: str, fund_holdings_table_name: str
+) -> None:
+    """
+    Loads fund data from the specified input data path and stores it in the database.
+
+    Args:
+        database_path (str): The path to the database.
+        input_data_path (str): The path to the input data files.
+        fund_holdings_table_name (str): The name of the table to store fund holdings data.
+
+    Returns:
+        None
+    """
+    logger = logging.getLogger()
+    df_list = []
+    files = [
+        f
+        for f in os.listdir(input_data_path)
+        if os.path.isfile(os.path.join(input_data_path, f)) and f.endswith(".csv")
+    ]
+    if not files:
+        logger.info("No CSV files found in the input data path.")
+        return
+    for file in files:
+        fund_name, date = parse_filename(file)
+        df = append_metadata(os.path.join(input_data_path, file), fund_name, date)
+        df_list.append(df)
+    if not df_list:
+        logger.info("No valid fund data found in the input data path.")
+        return
```

Comment:
Is this not an error condition?

Reply:
Indeed, since this the opposite condition to the block below (why not `else`?), this whole block could disappear if you are convinced it is not an error.